### ISSUE TITLE
confd does not reсover after etcd was temporary unavailable

### ISSUE THREAD

**Issue by aleks-v-k** [projectcalico/calicoctl#1179]
If etcd is unavailable for confd for some reason, then confd first writes in the log:

```
2016-09-29T08:32:44Z kbn3.local2 confd[32]: ERROR unexpected end of JSON input
2016-09-29T08:32:44Z kbn3.local2 confd[32]: ERROR unexpected end of JSON input
2016-09-29T08:32:44Z kbn3.local2 confd[32]: ERROR unexpected end of JSON input
2016-09-29T08:32:44Z kbn3.local2 confd[32]: ERROR unexpected end of JSON input
2016-09-29T08:32:44Z kbn3.local2 confd[32]: ERROR unexpected end of JSON input
2016-09-29T08:32:44Z kbn3.local2 confd[32]: ERROR unexpected end of JSON input
```

After this it will begin write to log infinitely

```
ERROR invalid character '<' looking for beginning of value
```

and this will not finish even after etcd become alive.
The simplest way to reproduce - stop etcd for a few seconds, and start it again.

Confd from upstream (tested with https://github.com/kelseyhightower/confd/releases/tag/v0.12.0-alpha3) does not have this problem. When etcd is down for confd there are records in the log: `ERROR client: etcd cluster is unavailable or misconfigured`, and it stops to write these errors when etcd become available.

**Comment by robbrockbank** [comment]
Additional info:
-  I tried to reproduce this locally and whilst I did hit the "Unexpected end of JSON input" error, things always seemed to recover once I'd started etcd again.  I was running etcd in a container on the same host as running calico/node container - perhaps related to that.

-  I tried swapping in the latest 0.12.1 confd from bacongobbler and seem to be hitting errors in our STs.  A little digging suggests that the config files generated from the confd templates seemed to be missing some etcd updates - so it looks as though there may be a timing window with the latest code.  Applying another etcd change to trigger confd then fixes things.
-  I also tried swapping in the latest etcd backend in confd v0.12.1 into the v0.10.0 confd codebase and also hit the same issue.  So looks as though the latest etcd backend in confd has some issues.
